polynote-server/src/main/scala/polynote/server/SharedNotebook.scala

```diff
@@ -214,11 +216,12 @@ class IOSharedNotebook(
       }
     }(_ => kernelLock.release)
 
+    // we want to queue all the cells, but evaluate them in order. So the outer IO of the result runs the outer IO of queueCell for all the cells.
```

Comment:
Thanks for this comment, it was really helpful in understanding what was going on here 😄